### PR TITLE
Add sample guide for the reference of "How to Extend Ballerina"

### PR DESCRIPTION
**Pull request by ldclakmal**
## Purpose
Purpose of this PR is to have a sample guide for the reference of [How to Extend Ballerina](https://ballerina.io/learn/how-to-extend-ballerina/)

This guide will be refered via the "1. Single file version" mentioned below.

image

### CODE REVIEW

File: guide/twilio_sample.bal [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+    config.clientConfig.url = BASE_URL;
+    string username;
+    string password;
+
+    var usernameOrEmpty = config.clientConfig.auth.username;
+    match usernameOrEmpty {
+        string usernameString => username = usernameString;
+        () => {
+            error err;
+            err.message = "Username cannot be empty";
```
[the remark this answers is not in the document]
Comment (MaryamZi) [reply guide/twilio_sample.bal]: Same with 118, 119.

File: guide/twilio_sample.bal [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+}
+
+public function TwilioClient::getCallerActions() returns TwilioConnector {
+    return self.twilioConnector;
+}
+// =========== End of implementation of the Endpoint
+
+
+// =========== Implementation for Connector
+public function TwilioConnector::getAccountDetails() returns (Account|error) {
```
Comment (MaryamZi) [inline guide/twilio_sample.bal]: Shall we remove the parentheses around the return type when it is a union type (here and in other similar functions)? 
```bal
public function TwilioConnector::getAccountDetails() returns Account|error
```

As a practice we use parentheses with return types only when required with tuple types.

File: guide/twilio_sample.bal (hunk 1 above)
[the remark this answers is not in the document]
Comment (ldclakmal) [reply guide/twilio_sample.bal]: Fixed.

File: guide/twilio_sample.bal (hunk 2 above)
Comment (ldclakmal) [reply guide/twilio_sample.bal]: Thanks. Fixed.

**Pull request merged**